website/blog/2021-02-22-jest-27.md

```diff
@@ -0,0 +1,73 @@
+For this reason, we are **changing the default test environment** from `"jsdom"` to `"node"`. If you are affected by this change because you use DOM APIs and do not have the test environment explicitly configured, you should be receiving an error when e.g. `document` is accessed, and you can configure `"testEnvironment": "jsdom"` or use [per-file environment configuration](https://jestjs.io/docs/en/configuration#testenvironment-string) to resolve this.  
+In the next major, we can now also eliminate `jest-jasmine2` and `jest-environment-jsdom` from the Jest dependency tree and require them to be installed explicitly, so that many users can profit from a smaller install size with less clutter that they don't need.
+
+Another default that we are changing affects Fake Timers aka [Timer Mocks](https://jestjs.io/docs/en/timer-mocks). We introduced an opt-in "modern" implementation of Fake Timers in Jest 26 accessed transparently through the same API, but with much more comprehensive mocking, such as for `Date` and `queueMicrotask`.  
+This **modern fake timers implementation will now be the default**. If you are among the unlucky few who are affected by the subtle implementation differences too heavily to migrate, you can get back the old implementation using `jest.useFakeTimers("legacy")` or, if you are enabling fake timers globally via [configuration](https://jestjs.io/docs/en/configuration#timers-string), `"timers": "legacy"`.
+
+## Features coming with breaking changes
+
+We introduced a few more small breaking changes to help you avoid mistakes by disallowing some things that can easily happen unintentionally:
+
+- The same `done` test callback may not be called more than once,
+- calling `done` and returning a Promise may not be combined,
+- a `describe` block must not return anything,
+
+and we [made some TypeScript types stricter](https://github.com/facebook/jest/pull/10512).
+
+Modules used in the following configuration options are now transformed like the rest of your code, which may be breaking if you relied on them being loaded as-is:
+
+- `testEnvironment`
+- `runner`
+- `testRunner`
+- `snapshotResolver` <!-- TODO pending merge https://github.com/facebook/jest/pull/8829 -->
+
+## Miscellaneous breaking changes
+
+We removed some long-deprecated functions:
+
+- `jest.addMatchers` (use `expect.extend` instead)
+- `jest.resetModuleRegistry` (use `jest.resetModules` instead)
+- `jest.runTimersToTime` (use `jest.advanceTimersByTime` instead)
+
+A lot of Jest's packages have been migrated to use ESM exports, so if you consume them directly, you may need to adjust your imports.
```

Comment:
Are these changes expected to cause many `.default` problems @SimenB? I didn't follow closely. If so I should perhaps expand a bit on this

Reply:
Nah, I don't think so. Can ask people to check the changelog for modules they use (`pretty-format` and `jest-diff` comes to mind)?

Reply:
But basically, for people who do `const format = require('pretty-format'); format(42);`, they will need to add a `.default` or `interopRequireDefault` or the likes, right? Just wanna make sure I understand the implication correctly

Reply:
yep!

Specifically for  `pretty-format` one can do `const {format} = require('pretty-format')`, but in general the diff will mostly be adding an `.default`

Reply:
I'll clarify a little